src/data/matrix/basic.lean

```diff
@@ -281,6 +377,16 @@ by simp [dot_product, add_mul, finset.sum_add_distrib]
   dot_product u (v + w) = dot_product u v + dot_product u w :=
 by simp [dot_product, mul_add, finset.sum_add_distrib]
 
+lemma dot_product_block' [has_mul α] [add_comm_monoid α] (v w : m ⊕ n → α) :
+dot_product v w = ∑ i, v (sum.inl i) * w (sum.inl i) + ∑ j, v (sum.inr j) * w (sum.inr j) :=
+by {rw [dot_product, ←fintype.sum_sum_elim], congr, ext (i | j); simp }
+
+lemma dot_product_block [has_mul α] [add_comm_monoid α] (v w : m ⊕ n → α) :
+dot_product v w =
+dot_product (λ i, v (sum.inl i))  (λ i, w (sum.inl i)) +
+dot_product (λ j, v (sum.inr j))  (λ j, w (sum.inr j)) :=
+by simp [dot_product, dot_product_block']
+
```

Comment:
I think these lemmas belong in `data/matrix/block.lean`

Reply:
Are you having a PR editing that file? If so, could you just copy the code there or I do it in this PR?